scipy/signal/filter_design.py

```diff
@@ -4843,6 +4845,206 @@ def _design_notch_peak_filter(w0, Q, ftype, fs=2.0):
+    fs : int, optional
+        The sampling frequency of the signal. `freq` must be between
+        0 and ``fs/2``. Default is 2.
+
+    Returns
+    -------
+    b, a : ndarray, ndarray
+        Numerator (``b``) and denominator (``a``) polynomials of the filter.
+
+    Raises
+    ------
+    ValueError
+        If `freq` is less than or equal to 0 or greater than or equal to
+        ``fs/2``, if `ftype` is not 'fir' or 'iir', if `order` is less than
+        or equal to 0 or greater than 24 when ``ftype='fir'``
+
+    See Also
+    --------
+    firwin
+    iirfilter
+
+    References
+    ----------
+    .. [1] Slaney, Malcolm, "An Efficient Implementation of the
+        Patterson–Holdsworth Auditory Filter Bank", Apple Computer
+        Technical Report 35, 1993, pp.3-8, 34-39.
+
+    Examples
+    --------
+    16-sample 4th order FIR Gammatone filter centered at 440 Hz
+
+    >>> from scipy import signal
+    >>> signal.gammatone(440, 'fir', numtaps=16, fs=16000)
+    (array([ 0.00000000e+00,  2.68586728e-07,  1.97702315e-06,  5.90239274e-06,
+        1.18022128e-05,  1.82536512e-05,  2.26825900e-05,  2.15748723e-05,
+        1.08452242e-05, -1.36746652e-05, -5.56777430e-05, -1.17807768e-04,
+       -2.01126526e-04, -3.04675651e-04, -4.25179658e-04, -5.56926298e-04]),
+       [1.0])
+
+    IIR Gammatone filter centered at 440 Hz
+
+    >>> from scipy import signal
+    >>> import matplotlib.pyplot as plt
+
+    >>> b, a = signal.gammatone(440, 'iir', fs=16000)
+    >>> w, h = signal.freqz(b, a)
+    >>> plt.plot(w / ((2 * np.pi) / 16000), 20 * np.log10(abs(h)))
+    >>> plt.xscale('log')
+    >>> plt.title('Gammatone filter frequency response')
+    >>> plt.xlabel('Frequency')
+    >>> plt.ylabel('Amplitude [dB]')
+    >>> plt.margins(0, 0.1)
+    >>> plt.grid(which='both', axis='both')
+    >>> plt.axvline(440, color='green') # cutoff frequency
+    >>> plt.show()
+    """
+    # Converts freq to float
+    freq = float(freq)
+
+    # Set sampling rate if not passed
+    if fs is None:
+        fs = 2
+    else:
+        fs = operator.index(fs)
+
+    # Check for invalid cutoff frequency or filter type
+    filter_types = ['fir', 'iir']
+    if freq <= 0 or freq >= fs / 2:
+        raise ValueError("The frequency must be between 0 and {}"
+                         " (nyquist), but given {}.".format(fs / 2, freq))
+    if ftype.lower() not in filter_types:
+        raise ValueError('ftype must be either fir or iir.')
+
+    # Calculate FIR gammatone filter
+    if ftype == 'fir':
+        # Set order and numtaps if not passed
+        if order is None:
+            order = 4
+        else:
+            order = operator.index(order)
+
+        if numtaps is None:
+            if fs <= 1000:
+                numtaps = 15
+            else:
+                numtaps = int(fs * 0.015)
+        else:
+            numtaps = operator.index(numtaps)
+
+        # Check for invalid order
+        if order <= 0 or order > 24:
```

Comment:
Perhaps also more readable
```suggestion
        if not 0 < order <= 24:
```

Reply:
Fixed. Also fixed `freq` between 0 and `fs/2`.